credential.js

```diff
@@ -146,22 +146,33 @@ var crypto = require('crypto'),
    *
    * Compare two strings, x and y with a constant-time
    * algorithm to prevent attacks based on timing statistics.
-   * 
+   *
+   * This really ought to be in C; see:
```

Comment:
> Implementing it in JS or C++ doesn't make much difference for the things I described above. You're going to end up dealing with JS objects either way and then it becomes nearly impossible to guarantee constant-time operation.

I think it may be better to do this using a typed array -- at least then we have a good understanding of what we're dealing with.